ci-operator/step-registry/ipi/conf/azure/resourcegroup/assign-role/ipi-conf-azure-resourcegroup-assign-role-commands.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# save the exit code for junit xml file generated in step gather-must-gather
+# pre configuration steps before running installation, exit code 100 if failed,
+# save to install-pre-config-status.txt
+# post check steps after cluster installation, exit code 101 if failed,
+# save to install-post-check-status.txt
+EXIT_CODE=100
+trap 'if [[ "$?" == 0 ]]; then EXIT_CODE=0; fi; echo "${EXIT_CODE}" > "${SHARED_DIR}/install-pre-config-status.txt"' EXIT TERM
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+cluster_rg=$(yq-go r ${CONFIG} 'platform.azure.resourceGroupName')
+vnet_rg=$(yq-go r ${CONFIG} 'platform.azure.networkResourceGroupName')
+
+if [[ -z "${cluster_rg}" ]] && [[ -z "${vnet_rg}" ]]; then
+    echo "This step used to grant proper permissions on scope of cluster rg or vnet rg, but both rg are empty, abort..."
+    exit 1
```

Comment:
hm should this be `exit 1` or `exit 0` (no error)?  I would have thought `exit 0` because there may be valid jobs that don't set resource group or particularly the network rg?

Reply:
Currently, this step only inserts into specific install workflow to cover one QE test case, I was thinking that is `exit 1` if both rgs are empty to make sure the step be performed in expected workflow.
But if it add into other valid jobs in the future on purpose, that make sense to be `exit 0` here, updated.